File: sentimental-cash/cash.py
# How many coins should i get back as a change

from cs50 import get_float


def main():
    while True:
        owed = get_float("Change owed in dollars: ")
        if owed > 0:
            break

    # QUARTER = 0.25
    # DIME = 0.10
    # NICKEL = 0.05
    # PENNY = 0.01

    coins= {'QUARTER': 0.25, 'DIME': 0.10, 'NICKEL': 0.05, 'PENNY': 0.01}

    quarters_number = int(owed / coins['QUARTER'])
    owed = round(owed - quarters_number * 0.25, 2)

    dimes_number = int(owed / coins['DIME'])
    owed = round(owed - dimes_number * 0.10, 2)

    nickels_number = int(owed / coins['NICKEL'])
    owed = round(owed - nickels_number * 0.05, 2)

    pennies_number = round(owed / coins['PENNY'])

    change = quarters_number + dimes_number + nickels_number + pennies_number
    print(change)


main()

File: cash/cash.c
// Check how much change in coins is due

#include <cs50.h>
#include <stdio.h>

#define QUARTER 25
#define DIME 10
#define NICKEL 5
#define PENNY 1

int calculate_coins(int cents, int one_coin_value);

int main(void)
{
    // ask for change owed in cents
    int n;
    do
    {
        n = get_int("Change owed in cents: "); // our owed value
    }
    while (n < 0);

    int quarters_number = calculate_coins(n, QUARTER);
    n = n - (quarters_number * QUARTER);

    int dimes_number = calculate_coins(n, DIME);
    n = n - (dimes_number * DIME);

    int nickels_number = calculate_coins(n, NICKEL);
    n = n - (nickels_number * NICKEL);

    int pennies_number = calculate_coins(n, PENNY);

    int change = quarters_number + dimes_number + nickels_number + pennies_number;
    printf("%i \n", change);
}

int calculate_coins(int cents, int one_coin_value)
{
    int change = 0;
    {
        change = cents / one_coin_value;
    }
    return change;
}
